spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/ErrorListener.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.support.ErrorMessage;
+
+/**
+ * Annotation that marks a method to consume {@link ErrorMessage}s from global Spring Integration errorChannel.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+@Target({ ElementType.METHOD, ElementType.ANNOTATION_TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@MessageMapping
+@Documented
+public @interface ErrorListener {
```

Comment:
I think this must be supplied with the `value` for any possible error channel. Even if with the `errorChannel` value by default.
What I mean is here in the `AbstractMessageChannelBinder`:
```
protected final ErrorInfrastructure registerErrorInfrastructure(ConsumerDestination destination, String group,
			C consumerProperties) {
	ErrorMessageStrategy errorMessageStrategy = getErrorMessageStrategy();
	ConfigurableListableBeanFactory beanFactory = getApplicationContext().getBeanFactory();
	String errorChannelName = errorsBaseName(destination, group, consumerProperties);
```

I'm pretty sure that end-user might want to be able to use this `@ErrorListener` for those channels as well.